editoast/src/views/infra/routes.rs

```diff
@@ -163,22 +166,142 @@ async fn get_routes_track_ranges<'a>(
     Ok(Json(result))
 }
 
+/// Return all routes that depend on the position of the specified nodes
+#[utoipa::path(
+    tag = "infra,routes",
+    params(InfraIdParam),
+    request_body(content = HashMap<String, Option<String>>, description = "A mapping node_id -> node_state | null"),
+    responses(
+        (status = 200, body = inline(Vec<String>), description = "A list of routes IDs")
+    ),
+)]
+#[get("/nodes")]
+async fn get_routes_nodes(
+    params: Path<InfraIdParam>,
+    infra_caches: Data<CHashMap<i64, InfraCache>>,
+    db_pool: Data<DbPool>,
+    Json(node_states): Json<HashMap<String, Option<String>>>,
+) -> Result<Json<Vec<String>>> {
+    let infra = match Infra::retrieve(db_pool.clone(), params.infra_id).await? {
+        Some(infra) => infra,
+        None => {
+            return Err(InfraApiError::NotFound {
+                infra_id: params.infra_id,
+            }
+            .into())
+        }
+    };
+
+    let mut conn = db_pool.get().await?;
+    let infra_cache = InfraCache::get_or_load(&mut conn, &infra_caches, &infra).await?;
+    let routes_cache = infra_cache.routes();
+
+    let result = routes_cache
+        .iter()
+        // We're only interested in routes that depend on specific node positions
+        .filter(|(_, route)| !route.unwrap_route().switches_directions.is_empty())
+        .filter(|(_, route)| {
+            let route = route.unwrap_route();
+            route
+                .switches_directions
+                .iter()
+                .all(
+                    |(node_id, node_state)| match node_states.get(node_id.as_str()) {
+                        // The route depends on the position of a node specified in the request, but maybe more
+                        Some(Some(required_state)) => required_state == node_state.as_str(),
+                        // The route depends on the position of a node specified in the request, but we allow
+                        // that node to be in any position (null value in the payload)
+                        Some(None) => true,
+                        // The route depend on the position of a node not specified in the request
+                        None => false,
+                    },
+                )
+        })
```

Comment:
To accept routes that depend on multiple nodes but don't have all of them specified in the request.

```suggestion
        .filter(|(_, route)| {
            let route = route.unwrap_route();
            node_states.iter().all(|(node_id, node_state)| {
                match route.switches_directions.get(&node_id.clone().into()) {
                    // The route crosses the requested node
                    Some(node_state_in_route) => {
                        if let Some(required_state) = node_state {
                            required_state == node_state_in_route.as_str()
                        } else {
                            true
                        }
                    }
                    // The route doesn't cross the requested node
                    None => false,
                }
            })
        })
```